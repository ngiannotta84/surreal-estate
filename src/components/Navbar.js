import React from "react";
import { Link } from "react-router-dom";
import "../styles/navbar.css";
import logo from "../assets/logo.png";

const Navbar = () => {
  return (
    <div className="navbar">
      <img className="logo" src={logo} alt="logo of Surreal Estate" />
      <ul className="navbar-links">
        <li className="navbar-links-item">
          <Link className="item" to="/">
            View Property
          </Link>
        </li>
        <li className="navbar-links-item">
          <Link className="item" to="add-property">
            Add a Property
          </Link>
        </li>
      </ul>
    </div>
  );
};
export default Navbar;
