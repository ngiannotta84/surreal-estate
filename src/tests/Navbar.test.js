import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import Navbar from "../components/Navbar";

describe("Navbar", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders view property", () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    const text = screen.getByText(/view property/i);
    expect(text).toBeInTheDocument();
  });
  it("renders the logo image", () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    const logo = screen.getByTestId("logo");
    expect(logo).toHaveClass("logo");
  });
});
