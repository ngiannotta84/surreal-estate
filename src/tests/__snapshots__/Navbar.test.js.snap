// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <img
      alt="logo of Surreal Estate"
      class="logo"
      data-testid="logo"
      src="logo.png"
    />
    <ul
      class="navbar-links"
    >
      <li
        class="navbar-links-item"
      >
        <a
          class="item"
          href="/"
        >
          View Property
        </a>
      </li>
      <li
        class="navbar-links-item"
      >
        <a
          class="item"
          href="/add-property"
        >
          Add a Property
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
